### DISCUSSION TITLE
Custom 404 page redirect

### DISCUSSION THREAD

**Discussion by AgentPrus** [vercel/next.js#20099] [Help]
Hi! I want to always redirect a user to the home page instead of the 404 page. What is the best possible way to handle it?

**Comment by ghost** [comment]
You can achieve this using [redirects](https://nextjs.org/docs/api-reference/next.config.js/redirects) in `next.config.js`.

```js
module.exports = {
  async redirects() {
    return [
      {
        source: '/404',
        destination: '/',
        permanent: true,
      },
    ]
  },
}
```

**Reply by AgentPrus** [reply]
It's doesn't work. Did you try it?

**Reply by ghost** [reply]
You can also redirect based on the data fetched in [getStaticProps](https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation).

```js
export async function getStaticProps(context) {
  const res = await fetch(`https://...`)
  const data = await res.json()

  if (!data) {
    return {
      redirect: {
        destination: '/',
        permanent: false,
      },
    }
  }

  return {
    props: {}, // will be passed to the page component as props
  }
}
```

In this case, if not data is fetched it will redirect home. This becomes more useful with dynamic routing.

**Reply by taoqf** [reply]
I tried redirecting in getStaticProps with no luck. It will throw an error during bulding. and direct me to https://github.com/vercel/next.js/blob/master/errors/gsp-redirect-during-prerender.md.
redirects works for me.
I tried exportPathMap with no luck after build or export.

@AgentPrus  This may work for your case:

```ts
index.getInitialProps = ({ req, res }) => {
	const url = '/pg001';
	res.statusCode = 302;
	const body = `<p>HTTP Redirect: 302. Redirecting to <a href="${url}">${url}</a></p>`;
	res.setHeader('Content-Length', Buffer.byteLength(body));
	res.setHeader('Location', url);

	if (req.method === 'HEAD') {
		res.end();
	} else {
		res.end(body);
	}
	return {};
};
```
